### ISSUE TITLE
Is there a way to compensate for small batch size

### ISSUE THREAD

**Issue by agenius5** [google-research/pegasus#139]
The paper uses a batch size of 256 but I can't do that. I can go to a max of 32 batch size. Batch size does affect the performance of the model so is there a way to compensate for smaller batch size by tuning learning rate or any other parameter?

Also the dataset registry in `public_param.py` defines batch size of 8 only with different learning rate and steps to train than reported in paper. So, are they adjusted for smaller batch size?

Last thing, the 32000 checkpoint for aeslc is as good as the one reported in paper (with 16k steps and different hyperparameter)?

**Comment by JingqingZ** [comment]
> The paper uses a batch size of 256 but I can't do that. I can go to a max of 32 batch size. Batch size does affect the performance of the model so is there a way to compensate for smaller batch size by tuning learning rate or any other parameter?

Experiments will be needed on this. May need more steps and smaller lr.
 
> Also the dataset registry in `public_param.py` defines batch size of 8 only with different learning rate and steps to train than reported in paper. So, are they adjusted for smaller batch size?

`public_param.py` was modified with smaller batch size so that local GPUs can run inference.

> Last thing, the 32000 checkpoint for aeslc is as good as the one reported in paper (with 16k steps and different hyperparameter)?

They should be comparable.

**Comment by agenius5** [comment]
@JingqingZ Thanks for your response.

I experimented with a small batch size and a new training method. Here are the results that I achieved on AESLC full datasets.

R1-F/R2-F/RL-F = `37.18/20.54/36.34`

I got these results with 8200 steps. The results published in paper were `37.68/21.25/36.51` with 16k steps. I am still trying to improve my results.

As you said earlier you won't be able to help but what do you think of these results?

**Comment by JingqingZ** [comment]
Yep, the results are comparable.
